src/actions/websocket.js

```diff
@@ -552,27 +553,15 @@ function handleChannelDeletedEvent(msg, dispatch, getState) {
     }
 }
 
-function handleChannelUpdatedEvent(msg, dispatch, getState) {
-    let channel;
-    try {
-        channel = msg.data ? JSON.parse(msg.data.channel) : null;
-    } catch (err) {
-        return;
-    }
+async function handleChannelUpdatedEvent(msg, dispatch, getState) {
+    const channelId = msg.data.channel_id;
+    const {data: channel} = await dispatch(getChannel(channelId));
 
-    const entities = getState().entities;
-    const {currentChannelId} = entities.channels;
-    if (channel) {
-        dispatch({
-            type: ChannelTypes.RECEIVED_CHANNEL,
-            data: channel,
-        });
-
-        if (currentChannelId === channel.id) {
-            // Emit an event with the channel received as we need to handle
-            // the changes without listening to the store
-            EventEmitter.emit(WebsocketEvents.CHANNEL_UPDATED, channel);
-        }
+    const {currentChannelId} = getState().entities.channels;
+    if (currentChannelId === channelId) {
```

Comment:
This seems racey: what happens if the `currentChannelId` changes before the `await` completes above?

Reply:
If the user switches to a different channel then there's no need to update the user with new channel info (in case of [mobile RN](https://github.com/mattermost/mattermost-mobile/blob/master/app/components/channel_drawer/channel_drawer.js#L82) - updating display name is not necessary).